LabManual.md

```diff
@@ -329,8 +329,481 @@ Click on "Train & Test" in the left sidebar.  Check the "Enable published model"
+
+> Fun Aside: The Suggested Utterances are extremely powerful.  LUIS makes smart decisions about which utterances to surface.  It chooses the ones that will help it improve the most to have manually labelled by a human-in-the-loop.  For example, if the LUIS model predicted that a given utterance mapped to Intent1 with 47% confidence and predicted that it mapped to Intent2 with 48% confidence, that is a strong candidate to surface to a human to manually map, since the model is very close between two intents.  
+
+Now that we can use our LUIS model to figure out the user's intent, let's integrate Azure search to find our pictures.  
 
 ### Configure for Azure Search ###
 
-TODO: add stuff to Web.config
+First, we need to provide our bot with the relevant information to connect to an Azure Search index.  The best place to store connection information is in the configuration file.  
+
+Open Web.config and in the appSettings section, add the following:
+
+```xml    
+    <!-- Azure Search Settings -->
+    <add key="SearchDialogsServiceName" value="" />
+    <add key="SearchDialogsServiceKey" value="" />
+    <add key="SearchDialogsIndexName" value="images" />
+```
+
+Set the value for the SearchDialogsServiceName to be the name of the Azure Search Service that you created earlier.  If needed, go back and look this up in the [Azure portal](https://portal.azure.com).  
+
+Set the value for the SearchDialogsServiceKey to be the key for this service.  This can be found in the [Azure portal](https://portal.azure.com) under the Keys section for your Azure Search.  In the below screenshot, the SearchDialogsServiceName would be "aiimmersionsearch" and the SearchDialogsServiceKey would be "375...".  
+
+![Azure Search Settings](./assets/AzureSearchSettings.jpg) 
+
+### Update Bot to use Azure Search ###
+
+Now, let's update the bot to call Azure Search.  First, open Tools-->NuGet Package Manager-->Manage NuGet Packages for Solution.  In the search box, type "Microsoft.Azure.Search".  Select the corresponding library, check the box that indicates your project, and install it.  It may install other dependencies as well.  
+
+![Azure Search NuGet](./assets/AzureSearchNuGet.jpg) 
+
+Right-click on your project in the Solution Explorer of Visual Studio, and select Add-->New Folder.  Create a folder called "Models".  Then right-click on the Models folder, and select Add-->Existing Item.  Do this twice to add these two files under the Models folder (make sure to adjust your namespaces if necessary):
+1. [ImageMapper.cs from the Finished section](./Finished/PictureBot/Models/ImageMapper.cs)
+2. [SearchHit.cs from the Finished section](./Finished/PictureBot/Models/SearchHit.cs)
+
+Next, right-click on the Dialogs folder in the Solution Explorer of Visual Studio, and select Add-->Class.  Call your class "SearchDialog.cs".  Add this code:
+
+```csharp
```

Comment:
Curious, do you find having them copy-paste code from the lab manual to be a more effective teaching tool than having that code already in the boilerplate and walking them through it? Haven't done this before, so honestly curious.